### PR TITLE
Topic improvments

### LINKED ISSUE #10: Topic / who set the topic?
[21:25:22] <prawnsalad> topic - who set it, could you add that into the issue tracker so dont forget?

Also to show the data when is set - dd/mm/year by who

### LINKED ISSUE #149: /topic #chan NOT showing who set the topic and when it was set
The /topic #channel command does NOT show when a topic was set and who set it. As the command normally does show the person who set it and when in other clients, e.g. mIRC, not sure if this is a bug or intentional. This issue is confirmed to take effect on InspIRCd 2.0.25 and at least one other instance of InspIRCd 2.0.

### PR DESCRIPTION
**Pull request by ItsOnlyBinary**
fixes #149
fixes #10

### CODE REVIEW

File: src/libs/IrcClient.js [hunk 1]
```diff
@@ -889,9 +891,28 @@ function clientMiddleware(state, network) {
...
+            let buffer = state.getOrAddBufferByName(networkid, event.channel);
+            buffer.topic_by = event.nick;
+            buffer.topic_when = event.when * 1000;
+
+            // add setby to the most recent topic component
+            let messages = buffer.getMessages();
+            for (let i = messages.length - 1; i >= 0; i--) {
+                let message = messages[i];
+                if (message.type === 'topic') {
+                    message.template = message.template.extend({
```
Comment (ItsOnlyBinary) [inline src/libs/IrcClient.js]: this is causing reactivity issues.

File: src/libs/IrcClient.js [hunk 2]
```diff
@@ -889,9 +891,28 @@ function clientMiddleware(state, network) {
...
+                        data() { return { topic_by: event.nick, topic_when: buffer.topic_when }; },
+                    });
```
Comment (ItsOnlyBinary) [inline src/libs/IrcClient.js]: this is causing reactivity issues.

**Comment by AllanOcelot** [comment]
This looks good to me! I'd just change the padding and font size 👍 

`.kiwi-message-topic {
    padding: 10px 20px;
    font-size: 1em;
}`

**Comment by ItsOnlyBinary** [comment]
Superseded by #882

**Pull request closed without merging**